feat: add versioning doc (#98)
beclab/docs: docs/developer/install/versioning.md

  - **Release cadence**: Based on testing status
  - **Examples**: `v1.11.0.rc.0`, `v1.11.0.rc.1`

- **Dailybuild releases**
  - Automatically generated from the `main` branch every day at 2:00 AM (Beijing Time).
  - Intended for development and testing; may be unstable.
  - **Release cadence**: Daily
  - **Examples**: `v1.12.0-20241201`

## Release branch management

During the `1.x` phase, Olares follows a structured monthly release cadence:

1. At the end of each month, a release branch (e.g., `release-1.11`) is created from the `main` branch.

2. An initial RC version (e.g., `v1.11.0.rc.0`) is released from the new release branch. Additional RC versions may follow as testing progresses.

3. The `main` branch transitions to the next version (e.g., from `v1.11` to `v1.12`).

4. Issues identified in the stable version are addressed through patch releases (e.g., `v1.11.1`), based on the corresponding release branch.

Developers can submit pull requests (PRs) to both the `main` branch and the relevant release branch as needed.

## Upgrade policies and compatibility

> During the `1.x` phase, Olares is in rapid iteration. **Automatic upgrades are not supported across different minor versions**. The upgrade and compatibility rules are as follows:
> 
> - **Within the same minor version**:  
>   Patch versions within the same minor version (e.g., `1.4.0` to `1.4.2`) are fully compatible and support automatic upgrades.
> 
> - **Across minor versions**:
>   Different minor versions (e.g., `1.4.x` to `1.5.x`) are not compatible. Manual uninstallation of the current version is required before installing the newer version.
> 
> - **Upgrade paths by version type**
>   - **Dailybuild**: Can upgrade to newer Dailybuild, RC, or stable versions.
>   - **RC**: Can upgrade to newer RC or stable versions.
>   - **Stable**: Can only upgrade to newer stable versions.

Reviewer: ```suggestion
Olares is in a rapid development phase with frequent feature additions and changes. To ensure stability and avoid unexpected issues, Olares follows a structured upgrade policy:
- **Upgrades within the same minor version**:  
   Upgrading within the same minor version (e.g., `1.4.0` to `1.4.2`) is fully supported. These updates typically include bug fixes or small improvements that do not affect compatibility.

- **Upgrades across minor versions**:  
   Upgrading from one minor version to another (e.g., `1.4.x` to `1.5.x`) involves more significant changes, such as new features or architectural updates. Because these changes may not be backward-compatible, automatic upgrades are **not supported**. Instead, users must manually uninstall the existing version before installing the newer version.

The type of release you are using also determines what upgrades are possible:
 - **Stable**: You can only upgrade to newer stable releases, ensuring maximum reliability and stability.
 - **RC**: You can upgrade to newer RC versions or stable releases as they become available.
 - **Daily build**: You can upgrade to newer Dailybuild, RC, or stable releases.
```

::: tip Future upgrade policy

> Olares plans to fully adopt semantic versioning once the major version exceeds `1`. This will enable seamless automatic upgrades across all versions within the same major version.

Reviewer: ```suggestion
In the next major release, Olares plans to fully support seamless automatic upgrades for all versions within the same major version.
```
